app/views/catalog/_document_list.html.erb

```diff
@@ -1,5 +1,11 @@
 <% # container for all documents in index list view -%>
 <% view_config = local_assigns[:view_config] || blacklight_config&.view_config(document_index_view_type) %>
 <div id="documents" class="documents-<%= view_config&.key || document_index_view_type %>">
-  <%= render documents, as: :document, view_config: view_config %>
+  <% view_config.document_component.with_collection(documents, counter_offset: (@response&.start || 0), presenter: Proc.new { |doc| document_presenter(doc) }).each do |component| %>
```

Comment:
Why is presenter a proc and not just a reference?

Reply:
(it's a reference in non-collection contexts)